src/Dialect/ONNX/ONNXOps/Canonicalize.cpp

```diff
@@ -2334,6 +2403,13 @@ void ONNXSpaceToDepthOp::getCanonicalizationPatterns(
   results.insert<RemoveSpaceToDepthDepthToSpacePattern>(context);
 }
 
+/// on the ONNXSplitOp
+void ONNXSplitOp::getCanonicalizationPatterns(
+    RewritePatternSet &results, MLIRContext *context) {
+  results.insert<PullReluLikeOpsThroughSplitPattern>(context);
```

Comment:
I wonder if it is easier to read if we are explicit here, say use a template instead:
```c
 results.insert<PullOpsThroughSplitPattern<ONNXReluOp>>(context);
 results.insert<PullOpsThroughSplitPattern<ONNXLeakyReluOp>>(context);
```
So that users don't need to look at `PullOpsThroughSplitPattern` to know which ops are supported. Also, there is no confusion that mix of ops is impossible.

Reply:
Thanks, thats a great suggestion